components/TextControls/src/Theming/MDCFilledTextField+MaterialTheming.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+#import "MDCFilledTextField.h"
+#import "MaterialContainerScheme.h"
+
+/**
+ This category is used to style MDCFilledTextField instances with an MDCContainerScheme.
+ */
+@interface MDCFilledTextField (MaterialTheming)
+
+/**
+ Applies a container scheme's subsystem-specific schemes to the receiver.
+
+ @param scheme A container scheme instance.
+ */
+- (void)applyThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme;
+
+/**
+ Applies a container scheme's subsystem-specific schemes to the receiver in a manner best suited to
+ convey an error state.
+
+ @param scheme A container scheme instance.
+ */
+- (void)applyErrorThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme;
```

Comment:
If a user enters _invalid_ text then they (the client) should re-apply theme?

Reply:
Yes. This was something @jverkoey @randallli and @romoore all felt very strongly about. The error "state" is not really a state, it's just a different application of a scheme. One benefit of this approach is it simplifies the implementation of overlapping "states," i.e. error coexisting with editing, resting, and disabled.